pkg/kubelet/apis/cri/v1alpha1/runtime/constants.go

```diff
@@ -25,3 +25,23 @@ const (
 	// NetworkReady means the runtime network is up and ready to accept containers which require network.
 	NetworkReady = "NetworkReady"
 )
+
+// StreamType is the type of the stream in CRI container log.
+type StreamType string
+
+const (
+	// Stdout is the stream type for stdout.
+	Stdout StreamType = "stdout"
+	// Stderr is the stream type for stderr.
+	Stderr StreamType = "stderr"
+)
+
+// LogTag is the tag of a log line in CRI container log.
+type LogTag string
+
+const (
+	// Partial means the line is part of multiple lines.
+	Partial LogTag = "P"
+	// End means the line is a single line or the end of multiple lines.
+	End LogTag = "E"
```

Comment:
IIUC, the log writer writes E whenever it gets newline from container?

Reply:
Yeah.